src/orquestra/sdk/_ray/_dag.py

```diff
@@ -238,16 +240,38 @@ class WfUserMetadata(t.TypedDict):
     workflow_def: t.Mapping[str, t.Any]
 
 
-def _get_python_packages(invocation: ir.TaskInvocation, wf: ir.WorkflowDef):
-    if not (deps_ids := wf.tasks[invocation.task_id].dependency_import_ids):
-        return []
+@singledispatch
+def _pip_string(_: ir.Import) -> t.List[str]:
+    return []
 
-    python_imports = [
-        imp
-        for imp_id in deps_ids
-        if isinstance((imp := wf.imports[imp_id]), ir.PythonImports)
+
+@_pip_string.register
+def _(imp: ir.PythonImports):
+    return [serde.stringify_package_spec(package) for package in imp.packages]
+
+
+@_pip_string.register
+def _(imp: ir.GitImport):
+    m = re.match(
+        r"(?P<user>.+)@(?P<domain>[^/]+?):(?P<repo>.+)", imp.repo_url, re.IGNORECASE
+    )
+    if m is not None:
+        url = f"ssh://{m.group('user')}@{m.group('domain')}/{m.group('repo')}"
+    else:
+        url = imp.repo_url
+    return [f"git+{url}@{imp.git_ref}"]
+
+
+def _import_pip_env(ir_invocation: ir.TaskInvocation, wf: ir.WorkflowDef):
+    task_def = wf.tasks[ir_invocation.task_id]
+    imports = [
+        wf.imports[id_]
+        for id_ in (
+            task_def.source_import_id,
+            *(task_def.dependency_import_ids or []),
+        )
     ]
-    return [package for imp in python_imports for package in imp.packages]
+    return list(itertools.chain(*(_pip_string(imp) for imp in imports)))
```

Comment:
**non-blocking**: nested list comprehension would be easier for me to interpret than `list()` + `chain()` + `*`

```suggestion
    return [chunk for imp in imports for chunk in _pip_string(imp)]
```